runner/app/routes/hardware.py

```diff
@@ -1,34 +1,64 @@
-from fastapi import APIRouter
-from pydantic import BaseModel
-from typing import Dict
 import os
+from typing import Dict
 
-from app.utils.hardware import get_cuda_devices, get_cuda_stats, HardwareDetail, HardwareStatDetail
+from app.utils.hardware import (
+    HardwareDetail,
+    HardwareStatDetail,
+    get_gpu_info,
+    get_gpu_stats,
+)
+from fastapi import APIRouter
+from pydantic import BaseModel
 
 router = APIRouter()
+
+
 class HardwareInformation(BaseModel):
+    """Response model for GPU information."""
+
     pipeline: str
     model_id: str
     gpu_info: Dict[int, HardwareDetail]
 
+
 class HardwareStats(BaseModel):
+    """Response model for real-time GPU statistics."""
     pipeline: str
     model_id: str
     gpu_stats: Dict[int, HardwareStatDetail]
 
-@router.get("/hardware/info", operation_id="hardware_info", response_model=HardwareInformation)
-async def hardware_info() -> HardwareInformation:
-    
+
+@router.get(
+    "/hardware/info",
+    operation_id="hardware_info",
+    response_model=HardwareInformation,
+)
+@router.get(
+    "/hardware/info/",
```

Comment:
Just curious, why do we need this in the routes? I know we have it on all the routes but have not been sure why we do when it looks like its excluded?

Reply:
This adds an extra layer of security to ensure that users receive the expected result, even if they include a trailing slash in the URL. A similar approach was implemented by Yondon in the [[/health endpoint](https://github.com/livepeer/ai-worker/blob/26950e6a1b5f1fc2c6e50b816df6746e5ae6515f/runner/app/routes/health.py#L12)](https://github.com/livepeer/ai-worker/blob/26950e6a1b5f1fc2c6e50b816df6746e5ae6515f/runner/app/routes/health.py#L12), so I followed the same logic here. However, since these are internal routes, this isn’t particularly critical.

I’m okay with leaving it out if you’d prefer.